### ISSUE TITLE
Emoji keyboard crash

### ISSUE THREAD

**Issue by stefanalund** [matrix-org/matrix-ios-sdk#4]
When running the sample app in the simulator I got the following crash:

```
2015-01-09 23:30:58.546 matrixConsole[19260:167120] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: '4246239423_iphone-emoji-control-keylayout: An -observeValueForKeyPath:ofObject:change:context: message was received but not handled.
Key path: center
Observed object: <UIInputSetHostView: 0x7fe3fac21c50; frame = (0 667; 375 258); layer = <CALayer: 0x7fe3fac2de60>>
Change: {
    kind = 1;
}
Context: 0x0'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000106382f35 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x0000000105915bb7 objc_exception_throw + 45
	2   CoreFoundation                      0x0000000106382e6d +[NSException raise:format:] + 205
	3   Foundation                          0x00000001052bf78f -[NSObject(NSKeyValueObserving) observeValueForKeyPath:ofObject:change:context:] + 73
	4   Foundation                          0x00000001051ee9a6 NSKeyValueNotifyObserver + 356
	5   Foundation                          0x00000001051edbcd NSKeyValueDidChange + 466
	6   Foundation                          0x00000001051f228f -[NSObject(NSKeyValueObserverNotification) didChangeValueForKey:] + 118
	7   UIKit                               0x0000000106b0ff04 -[UIView(Geometry) _applyISEngineLayoutValues] + 855
	8   UIKit                               0x0000000106b1001e -[UIView(Geometry) _resizeWithOldSuperviewSize:] + 150
	9   CoreFoundation                      0x00000001062bb042 __53-[__NSArrayM enumerateObjectsWithOptions:usingBlock:]_block_invoke + 114
	10  CoreFoundation                      0x00000001062ba76f -[__NSArrayM enumerateObjectsWithOptions:usingBlock:] + 255
	11  UIKit                               0x0000000106b0ea60 -[UIView(Geometry) resizeSubviewsWithOldSize:] + 143
	12  UIKit                               0x000000010712a091 -[UIView(AdditionalLayoutSupport) _is_layout] + 143
	13  UIKit                               0x0000000106b13ceb -[UIView(Hierarchy) _updateConstraintsAsNecessaryAndApplyLayoutFromEngine] + 560
	14  UIKit                               0x0000000106b1f973 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 521
	15  QuartzCore                          0x0000000106931de8 -[CALayer layoutSublayers] + 150
	16  QuartzCore                          0x0000000106926a0e _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 380
	17  UIKit                               0x0000000106b13847 -[UIView(Hierarchy) layoutBelowIfNeeded] + 611
	18  UIKit                               0x0000000107174a17 -[UIInputSetHostView layoutIfNeeded] + 105
	19  UIKit                               0x0000000106b175ce +[UIView(Animation) performWithoutAnimation:] + 65
	20  UIKit                               0x00000001071755d0 -[UIInputWindowController performWithoutAppearanceCallbacks:] + 28
	21  UIKit                               0x00000001071755fd -[UIInputWindowController performWithoutCallbacksOrNotifications:] + 35
	22  UIKit                               0x0000000107178b60 -[UIInputWindowController updateToPlacement:withNormalAnimationsAndNotifications:] + 176
	23  UIKit                               0x0000000106f6dfe5 -[UIInputViewAnimationControllerBasic prepareAnimationWithHost:startPlacement:endPlacement:] + 33
	24  UIKit                               0x0000000107177ecf -[UIInputWindowController moveFromPlacement:toPlacement:starting:completion:] + 590
	25  UIKit                               0x000000010717bdc2 -[UIInputWindowController setInputViewSet:] + 599
	26  UIKit                               0x0000000107177a7a -[UIInputWindowController performOperations:withAnimationStyle:] + 50
	27  UIKit                               0x0000000106f67bce -[UIPeripheralHost(UIKitInternal) setInputViews:animationStyle:] + 1054
	28  UIKit                               0x0000000106c1d31d -[UIResponder becomeFirstResponder] + 468
	29  UIKit                               0x0000000106b12e03 -[UIView(Hierarchy) becomeFirstResponder] + 99
	30  UIKit                               0x00000001071d9ad7 -[UITextField becomeFirstResponder] + 51
	31  UIKit                               0x0000000106e619c1 -[UITextInteractionAssistant(UITextInteractionAssistant_Internal) setFirstResponderIfNecessary] + 177
	32  UIKit                               0x0000000106e63a30 -[UITextInteractionAssistant(UITextInteractionAssistant_Internal) oneFingerTap:] + 2263
	33  UIKit                               0x0000000106e592e6 _UIGestureRecognizerSendActions + 262
	34  UIKit                               0x0000000106e57f89 -[UIGestureRecognizer _updateGestureWithEvent:buttonEvent:] + 532
	35  UIKit                               0x0000000106e5cba6 ___UIGestureRecognizerUpdate_block_invoke662 + 51
	36  UIKit                               0x0000000106e5caa2 _UIGestureRecognizerRemoveObjectsFromArrayAndApplyBlocks + 254
	37  UIKit                               0x0000000106e52b1d _UIGestureRecognizerUpdate + 2796
	38  UIKit                               0x0000000106aecff6 -[UIWindow _sendGesturesForEvent:] + 1041
	39  UIKit                               0x0000000106aedc23 -[UIWindow sendEvent:] + 667
	40  UIKit                               0x0000000106aba9b1 -[UIApplication sendEvent:] + 246
	41  UIKit                               0x0000000106ac7a7d _UIApplicationHandleEventFromQueueEvent + 17370
	42  UIKit                               0x0000000106aa3103 _UIApplicationHandleEventQueue + 1961
	43  CoreFoundation                      0x00000001062b8551 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	44  CoreFoundation                      0x00000001062ae41d __CFRunLoopDoSources0 + 269
	45  CoreFoundation                      0x00000001062ada54 __CFRunLoopRun + 868
	46  CoreFoundation                      0x00000001062ad486 CFRunLoopRunSpecific + 470
	47  GraphicsServices                    0x0000000109e189f0 GSEventRunModal + 161
	48  UIKit                               0x0000000106aa6420 UIApplicationMain + 1282
	49  matrixConsole                       0x0000000104d4cf23 main + 115
	50  libdyld.dylib                       0x0000000107fc8145 start + 1
	51  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException

```

**Comment by ara4n** [comment]
Weird; i can't reproduce this... what XCode version / iOS version / simulated device model was this?  (Screencap of me successfully sending emoji in the 4S simulator (8.1 (550.3)) with XCode 6.1 (6A1052d) on Mavericks.

![screen shot 2015-01-10 at 02 17 08](https://cloud.githubusercontent.com/assets/1294269/5690136/bea12118-986f-11e4-8096-0377581abd3d.png)

**Comment by stefanalund** [comment]
I could actually also input emojis. The crash occurred (I think) when navigating away from the chat. 

Running the latest versions available.

**Comment by ara4n** [comment]
Hopefully fixed in 7a5c5e1 (at least, i can't repro it any more).

**Comment by stefanalund** [comment]
👍
